### DISCUSSION TITLE
change the exposure settings while Night-Lapse Photo is running

### DISCUSSION THREAD

**Discussion by tomas123** [gopro/labs#500] [Show and tell]
You can change the settings **while** (!) Night-Lapse Photo is running.
While sundown we need the automatic mode for iso and speed.
After sunset, the Hero 11's auto exposure works very poorly. The starry sky is much too dark.
So I use David's EV formula to toogle the exposure in manual mode (ISO800 30s) at a level 0 EV.

`<l1t+i8M1+mNP+eA+p.30+r=E6.25=Gi=G*0.01=E/G=E*s=E#2<E0.1t+i8M8+mNP+p+e30+!X"         EV=$E        "!1N!R`

```
<l1  //if new loop starts set Sundown-Mode 
    t+i8M1        //ISO 100-800
    +mNP         // mode Night-Lapse Photo
    +eA+p.30     //Auto Shutter and Interval 30s (set eA before p.30!)
    +r           //RAW + JPG
=E6.25           //calc EV from https://gopro.github.io/labs/control/actions/
=Gi
=G*0.01
=E/G
=E*s
=E#2          // Hero11 minimum_EV=0.0495 calculated from 1/2.6 ISO1575
<E0.1  // if EV<0.1 then switch to Astro-Mode and Exit Script
    t+i8M8     //fix ISO=800
    +mNP+p+e30 //Shutter=30s, Intervall=Auto (ca. 32s)
    +!X    //exit Script
"         EV=$E        "   //print measured EV
!1N   //wait 1s
!R    //Repeat
```

 

First Scan QR Code and then start Night-Lapse Photo.
I use LRTimelapse to combine the RAW images into a video.
The Holy Grail Mode can compensate jumps in exposure.

**Comment by dnewman-gpsw** [comment]
I've been meaning to do that. Thanks for building this.

Your link didn't work.

https://gopro.github.io/labs/control/set/?cmd=%3Cl1t%2Bi8M1%2BmNP%2BeA%2Bp.30%2Br=E6.25=Gi=G0.01=E/G=Es=E%232%3CE0.1t%2Bi8M8%2BmNP%2Bp%2Be30%2B!X%22%20EV=$E%20%22!1N!R

![image](https://github.com/gopro/labs/assets/10856084/d9887b70-c735-4167-90fd-a6b5e3f7320f)

**Comment by tomas123** [comment]
surely this is a very special problem, but I give a feedback after a few test runs:

The EV metering of the Gopro is very erratic at values below EV 0.
Therefore I defined a moving average EV and wrote it into the EXIF header for evaluation.

With this setting I found an acceptable transition from auto-shutter to 30s-shutter at EV -6

`<l1t+i8M1+mNP+eA+p.30+r+=M50=E6.25=Gi=G*0.01=E/G=E*s=E#2<EM=M+E+=M/2<M-6t+i8M8+mNP+p+e30+!XoMEVEV="$E"oMEVMI="$M"" EV=$E\n min=$M "!1N!R`

With EV -6 the 30s recording is not overexposed in the twilight.

crop from GoPro JPG
Left: ISO 800, Last Image with Auto Shutter (3s)
Right the next Image: ISO 800, 30s Shutter
![compare](https://github.com/gopro/labs/assets/222947/9c5cdc6d-cb3e-4ae9-a0ea-432afc022a5a)

**it would be much easier if we could read the ISO and shutter from the last taken image**

with Exiftool we can read the own defined metadata from Exif Header
`% exiftool -u -GoPro_EVMI /Volumes/256GB/DCIM/106GOPRO/G058*.JPG
.
======== /Volumes/256GB/DCIM/106GOPRO/G0583319.JPG
GoPro EVMI                      : 1.5749
======== /Volumes/256GB/DCIM/106GOPRO/G0583320.JPG
GoPro EVMI                      : 1.5749
======== /Volumes/256GB/DCIM/106GOPRO/G0583321.JPG
GoPro EVMI                      : 1.5387
======== /Volumes/256GB/DCIM/106GOPRO/G0583322.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583323.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583324.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583325.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583326.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583327.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583328.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583329.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583330.JPG
GoPro EVMI                      : 1.533
======== /Volumes/256GB/DCIM/106GOPRO/G0583331.JPG
GoPro EVMI                      : 1.4868
======== /Volumes/256GB/DCIM/106GOPRO/G0583332.JPG
GoPro EVMI                      : 1.3911
======== /Volumes/256GB/DCIM/106GOPRO/G0583333.JPG
GoPro EVMI                      : 1.2986
======== /Volumes/256GB/DCIM/106GOPRO/G0583334.JPG
GoPro EVMI                      : 1.1791
======== /Volumes/256GB/DCIM/106GOPRO/G0583335.JPG
GoPro EVMI                      : 1.0131
======== /Volumes/256GB/DCIM/106GOPRO/G0583336.JPG
GoPro EVMI                      : 0.9325
======== /Volumes/256GB/DCIM/106GOPRO/G0583337.JPG
GoPro EVMI                      : 0.8332
======== /Volumes/256GB/DCIM/106GOPRO/G0583338.JPG
GoPro EVMI                      : 0.7429
======== /Volumes/256GB/DCIM/106GOPRO/G0583339.JPG
GoPro EVMI                      : 0.6516
======== /Volumes/256GB/DCIM/106GOPRO/G0583340.JPG
GoPro EVMI                      : 0.5747
======== /Volumes/256GB/DCIM/106GOPRO/G0583341.JPG
GoPro EVMI                      : 0.4903
======== /Volumes/256GB/DCIM/106GOPRO/G0583342.JPG
GoPro EVMI                      : 0.3937
======== /Volumes/256GB/DCIM/106GOPRO/G0583343.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583344.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583345.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583346.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583347.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583348.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583349.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583350.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583351.JPG
GoPro EVMI                      : -1.8649
======== /Volumes/256GB/DCIM/106GOPRO/G0583352.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583353.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583354.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583355.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583356.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583357.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583358.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583359.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583360.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583361.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583362.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583363.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583364.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583365.JPG
GoPro EVMI                      : -3.3332
======== /Volumes/256GB/DCIM/106GOPRO/G0583366.JPG
GoPro EVMI                      : -4.2501
======== /Volumes/256GB/DCIM/106GOPRO/G0583367.JPG
GoPro EVMI                      : -4.2501
======== /Volumes/256GB/DCIM/106GOPRO/G0583368.JPG
GoPro EVMI                      : -4.2501
======== /Volumes/256GB/DCIM/106GOPRO/G0583369.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583370.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583371.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583372.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583373.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583374.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583375.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583376.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583377.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583378.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583379.JPG
GoPro EVMI                      : -4.8151
======== /Volumes/256GB/DCIM/106GOPRO/G0583380.JPG
GoPro EVMI                      : -5.4038
======== /Volumes/256GB/DCIM/106GOPRO/G0583381.JPG
GoPro EVMI                      : -5.4038
======== /Volumes/256GB/DCIM/106GOPRO/G0583382.JPG
GoPro EVMI                      : -5.4038
======== /Volumes/256GB/DCIM/106GOPRO/G0583383.JPG
GoPro EVMI                      : -5.795
======== /Volumes/256GB/DCIM/106GOPRO/G0583384.JPG
GoPro EVMI                      : -5.795
======== /Volumes/256GB/DCIM/106GOPRO/G0583385.JPG
GoPro EVMI                      : -5.795
======== /Volumes/256GB/DCIM/106GOPRO/G0583386.JPG
GoPro EVMI                      : -5.795
======== /Volumes/256GB/DCIM/106GOPRO/G0583387.JPG
GoPro EVMI                      : -5.795
======== /Volumes/256GB/DCIM/106GOPRO/G0583388.JPG
GoPro EVMI                      : -5.795
`

**Reply by dnewman-gpsw** [reply]
This is very cool stuff.  Even using scripted custom metadata, I'm sure you are the first to do so. I want to dig into this more when back from traveling.
